unison-core/src/Unison/Referent.hs

```diff
@@ -19,17 +20,17 @@ import qualified Unison.ConstructorType as CT
 
 -- Slightly odd naming. This is the "referent of term name in the codebase",
 -- rather than the target of a Reference.
-type Referent = Referent' Reference
+type Referent = TermRef Reference
 pattern Ref :: Reference -> Referent
-pattern Ref r = Ref' r
+pattern Ref r = IdRef r
 pattern Con :: Reference -> Int -> ConstructorType -> Referent
-pattern Con r i t = Con' r i t
+pattern Con r i t = ConRef r i t
 {-# COMPLETE Ref, Con #-}
 
-type Id = Referent' R.Id
+type Id = TermRef R.Id
 
-data Referent' r = Ref' r | Con' r Int ConstructorType
-  deriving (Show, Ord, Eq, Functor)
+data TermRef r = IdRef r | ConRef r Int ConstructorType
+  deriving (Show, Ord, Eq, Functor, Generic)
 
```

Comment:
The names for this type and its constructors were pretty terrible, so I improved them a little.

Reply:
For better or worse, `Id` in the context of references has meant "statically known to not be a Builtin", which isn't the case for this `IdRef`, so I think this will be pretty confusing.  I guess if the previous names are intolerable, I would just switch them to `Ref` and `Con`, and give up on the patterns that preserved the original types.  I don't think `IdRef` is more meaningful than `Ref'`, is it?  

IIRC this nomenclature shows up in a few other places, so even if we decide to change it, I'd request that we revert this change for now, and then do a more comprehensive terminology switch after getting the codebase2 branch where we want it.

Reply:
+1 on reverting this and doing a terminology review at a more convenient time.

Reply:
The real issue is that `'` can't appear in a type that you serve out with Servant. So the solution is either:

1. Rename these things.
2. Copy-paste them into the HTTP module with different names.

I think 2 is the solution we'll have to end up with eventually.

Reply:
@runarorama Isn't there a third option: don't use the `Generic` autoderived JSON instance, just write a `From/ToJSON` manually (or whatever typeclasses you have to give to be served up by Servant).

Giving manual JSON instances (possibly as orphan instances in the Server packages) also is another way to  avoid having to pessimistically fork copies of every type that the server works with.